src/test/java/io/skai/okta/internshipstreamingserviceparserxml/services/impl/LostFilmServiceTest.java

```diff
@@ -0,0 +1,84 @@
+package io.skai.okta.internshipstreamingserviceparserxml.services.impl;
+
+import io.skai.okta.internshipstreamingserviceparserxml.dto.Episode;
+import io.skai.okta.internshipstreamingserviceparserxml.kafka.MessageProducer;
+import io.skai.okta.internshipstreamingserviceparserxml.repository.VideoRepository;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+
+@ExtendWith(MockitoExtension.class)
+class LostFilmServiceTest {
+
+    @InjectMocks
+    LostFilmService service;
+
+    @Mock
+    private VideoRepository repository;
+
+    @Mock
+    private MessageProducer producer;
+
+    private final Episode episode = Episode
+            .builder()
+            .title("test title")
+            .description("test description")
+            .pubDate(LocalDateTime.parse("Sun, 19 Dec 2021 18:50:04 +0000",
+                    DateTimeFormatter.ofPattern("E, d MMM yyyy HH:mm:ss Z")))
+            .link("test link")
+            .season(1)
+            .episodeNumber(1)
+            .tvSeriesTitle("test tv series title")
+            .build();
+
+    @Test
+    void testSaveEpisodeSuccessful() {
+        service.saveEpisode(episode);
+        service.saveEpisode(episode);
+        verify(repository, times(2)).saveOrUpdate(episode);
+        verify(producer, times(2)).produce(eq("omdb"), eq(episode));
+        /*
+        mockito couldn't check invocation of this methods in "saveOrUpdate()"?
```

Comment:
no because repo is mocked and this is correct, repository implementations should be verified in repo test class not in service